### PR TITLE
chore: add unit tests, fix bug for date marshalling

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
Adding unit tests

##### CDK / CloudFormation Parameters Changed
N/A

#### Issue #, if available
N/A

#### Description of how you validated changes
Unit tests

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-default-value-transformer/src/validators.ts [hunk 1]
```diff
@@ -17,8 +17,8 @@ const validateDate = (date: string): boolean => {
   // Verify the correct number of days for
   // the month contained in the date-string
   const year = Number(date.slice(0, 4));
-  const month = Number(date.slice(5, 2));
-  const day = Number(date.slice(8, 2));
+  const month = Number(date.slice(5, 7));
```
Comment (phani-srikar) [inline packages/amplify-graphql-default-value-transformer/src/validators.ts:20]: this looks like a bug that we're fixing here?

**Review by phani-srikar — COMMENTED** [review]
one question

File: packages/amplify-graphql-default-value-transformer/src/validators.ts (hunk 1 above)
Comment (alharris-at) [reply packages/amplify-graphql-default-value-transformer/src/validators.ts:20]: Yeah, I think somebody interpreted the slice api as (start, length), but it's (start, end). The effect here is that day/month were ALWAYS deserialized as 0, and we never did day value validation.

Comment (alharris-at) [reply packages/amplify-graphql-default-value-transformer/src/validators.ts:20]: It's tested in the AWSDate tests (e.g. ensuring that feb fails on 29 in some cases, on 30 in nov, etc.

Comment (alharris-at) [reply packages/amplify-graphql-default-value-transformer/src/validators.ts:20]: Previously those tests would always pass.

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**